analyzeme/src/stringtable.rs

```diff
@@ -87,6 +117,32 @@ impl<'st> StringRef<'st> {
             }
         }
     }
+
+    fn get_addr(&self) -> Result<Addr, ()> {
+        if self.id.is_virtual() {
+            match self.table.index.get(&self.id) {
+                Some(&addr) => Ok(addr),
+                None => Err(()),
+            }
+        } else {
+            Ok(self.id.to_addr())
+        }
+    }
+}
+
+fn is_utf8_continuation_byte(byte: u8) -> bool {
+    // See module-level documentation for more information on the encoding.
+    const UTF8_CONTINUATION_MASK: u8 = 0b1100_0000;
+    const UTF8_CONTINUATION_BYTE: u8 = 0b1000_0000;
+    (byte & UTF8_CONTINUATION_MASK) == UTF8_CONTINUATION_BYTE
+}
+
+// String IDs in the table data are encoded in big endian format, while string
+// IDs in the index are encoded in little endian format. Don't mix the two up.
```

Comment:
Why do we use two different formats? Is it because of the utf-8 stuff we're doing?

Reply:
Yes, we need big endian so that the uppermost bits end up in the first byte we encounter. But for the index table that's not necessary, so little endian is the more efficient option (at least on little endian platforms, which means all the platforms `:)`). We _probably_ could use big endian everywhere without noticing at real performance difference though.

Reply:
If we implemented https://github.com/rust-lang/measureme/issues/96 then this issue would solve itself too.